src/main/java/com/miliancode/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+import com.miliancode.mapper.CartItemMapper;
+import com.miliancode.mapper.ShoppingCartMapper;
+import com.miliancode.model.Book;
+import com.miliancode.model.CartItem;
+import com.miliancode.model.ShoppingCart;
+import com.miliancode.model.User;
+import com.miliancode.repository.BookRepository;
+import com.miliancode.repository.CartItemRepository;
+import com.miliancode.repository.ShoppingCartRepository;
+import com.miliancode.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addItemToShoppingCart(CreateCartItemRequestDto createCartItemRequestDto,
+                                                 Long userId) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        Book book = bookRepository.findById(createCartItemRequestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find book by id = " + createCartItemRequestDto.getBookId()));
+        shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().getId().equals(createCartItemRequestDto.getBookId()))
+                .findFirst()
+                .ifPresentOrElse(item -> item.setQuantity(
+                                item.getQuantity() + createCartItemRequestDto.getQuantity()),
+                        () -> addCartItem(shoppingCart, book, createCartItemRequestDto));
+        shoppingCartRepository.save(shoppingCart);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto updateCartItem(Long itemId,
+                                          UpdateCartItemRequestDto updateCartItemRequestDto,
+                                          Long userId) {
+        CartItem cartItem = cartItemRepository
+                .findByIdAndShoppingCartUserId(itemId, userId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Can't find shopping cart for user with id "
+                                        + userId
+                        )
+                );
+        cartItem.setQuantity(updateCartItemRequestDto.getQuantity());
+        cartItemRepository.save(cartItem);

```

Comment:
You don't need to call method "save()" when you update existing entities/records, because Hibernate will automatically detect this changes and perfrom update action. It's called "dirty checking" 
https://vladmihalcea.com/the-anatomy-of-hibernate-dirty-checking/
https://docs.jboss.org/hibernate/orm/6.6/introduction/html_single/Hibernate_Introduction.html